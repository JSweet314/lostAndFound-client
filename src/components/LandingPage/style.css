.landing-page {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.5);
  padding: 10rem 0 0 0;
}

.landing-page__btnGroup {
  display: flex;
  justify-content: space-evenly;
  margin: 0 0 3rem 0;
  width: 50%;
}

.landing-page__btn {
  padding: 1rem;
  background-color: rgba(0, 0, 0, 0.5);
  color: #fff;
  font-size: 3rem;  
  text-decoration: none;
  text-align: center;
  border-radius: 15px;
  box-shadow: 2px 2px 5px rgba(196, 126, 89, 0.5);
  width: 15rem;
}

.landing-page__btn:hover {
  box-shadow: 2px 2px 5px rgba(196, 126, 89, 1);
  background-color: rgba(0, 0, 0, 0.5);
  color: #C47E59;
}