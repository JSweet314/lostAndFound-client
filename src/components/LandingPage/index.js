import React from 'react';
import Welcome from '../Welcome';
import {Link} from 'react-router-dom';
import Search from '../Search';
import PropTypes from 'prop-types';
import './style.css';

const LandingPage = ({username}) => {
  return (
    <main className='landing-page'>
      <Welcome username={username} />
      {
        username &&
        <div className='landing-page__btnGroup'>
          <Link 
            className='landing-page__btn' 
            to='/report/lost'>
          Lost an Item?
          </Link> 
          <Link 
            className='landing-page__btn' 
            to='/report/found'>
          Found an Item?
          </Link> 
        </div>
      }
      <Search />
    </main>
  );
};

LandingPage.propTypes = {
  username: PropTypes.string
};

export default LandingPage;