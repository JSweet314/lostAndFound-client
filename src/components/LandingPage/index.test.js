import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from './index';

describe('LandingPage', () => {
  it('should match a snapshot when no user', () => {
    const wrapper = shallow(<LandingPage />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should match a snapshot when user is signed in', () => {
    const wrapper = shallow(<LandingPage username='jon'/>);
    expect(wrapper).toMatchSnapshot();
  });
});