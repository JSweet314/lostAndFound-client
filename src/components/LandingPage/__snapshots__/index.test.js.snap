// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match a snapshot when no user 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Welcome
          username={undefined}
/>,
        undefined,
      ],
      "className": "landing-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "username": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Welcome
            username={undefined}
/>,
          undefined,
        ],
        "className": "landing-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "username": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`LandingPage should match a snapshot when user is signed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage
    username="jon"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Welcome
          username="jon"
/>,
        <div
          className="landing-page__btnGroup"
>
          <Link
                    className="landing-page__btn"
                    replace={false}
                    to="/report/lost"
          >
                    LOST
          </Link>
          <Link
                    className="landing-page__btn"
                    replace={false}
                    to="/report/found"
          >
                    FOUND
          </Link>
</div>,
      ],
      "className": "landing-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "username": "jon",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="landing-page__btn"
              replace={false}
              to="/report/lost"
>
              LOST
</Link>,
            <Link
              className="landing-page__btn"
              replace={false}
              to="/report/found"
>
              FOUND
</Link>,
          ],
          "className": "landing-page__btnGroup",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "LOST",
              "className": "landing-page__btn",
              "replace": false,
              "to": "/report/lost",
            },
            "ref": null,
            "rendered": "LOST",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "FOUND",
              "className": "landing-page__btn",
              "replace": false,
              "to": "/report/found",
            },
            "ref": null,
            "rendered": "FOUND",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Welcome
            username="jon"
/>,
          <div
            className="landing-page__btnGroup"
>
            <Link
                        className="landing-page__btn"
                        replace={false}
                        to="/report/lost"
            >
                        LOST
            </Link>
            <Link
                        className="landing-page__btn"
                        replace={false}
                        to="/report/found"
            >
                        FOUND
            </Link>
</div>,
        ],
        "className": "landing-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "username": "jon",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="landing-page__btn"
                replace={false}
                to="/report/lost"
>
                LOST
</Link>,
              <Link
                className="landing-page__btn"
                replace={false}
                to="/report/found"
>
                FOUND
</Link>,
            ],
            "className": "landing-page__btnGroup",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "LOST",
                "className": "landing-page__btn",
                "replace": false,
                "to": "/report/lost",
              },
              "ref": null,
              "rendered": "LOST",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "FOUND",
                "className": "landing-page__btn",
                "replace": false,
                "to": "/report/found",
              },
              "ref": null,
              "rendered": "FOUND",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
